### DISCUSSION TITLE
Add support to overwrite only the parts of the root meta function

### DISCUSSION THREAD

**Discussion by felixaschultz** [remix-run/remix#9523] [Proposals]
After starting working with remix in projects I fell in love with it. It's a great framework to build modern web apps.
While I´ve worked on a big project with remix, I´ve found out by an accident that the root meta function was overwritten, after I exported on an route a  meta function, where I´ve just added a new title. My first thought was that only the title where overwritten in the root file and not the robots object.

root.jsx
`export function meta() {
  return [
    { title: "Candidate.dk" },
    { robots: "none" },
  ];
}`

route.jsx
`export function meta() {
  return [
    { title: "Candidate.dk — Candidates for Digital Internships and Jobs" },
  ];
}`

It would be an nice idea if the root meta function and maybe also the links function where global settings for a site, while meta and link function on routes would be children for the root meta and links function, and only replace in the root the objects that where defined in the route meta function.

So in the example above only the title would be changed in the root and the robots object wouldn´t be touched or overwritten.

**Comment by sergiodxa** [comment]
In v1, the MetaFunction used an object and automatically merged the meta keys, this way if you only overwrote the title only that changed, this had issues where some meta keys could be added multiple times (e.g. `og:author`) but an object only allows one value per key, so the API had to support those cases by making a more complex API.

In Remix v2 this was changed to use an array, your MetaFunction now receives the parent route array of meta tags, this way you could grab the list, map it and change only the title if you want.

There's also `@remix-run/v1-meta` package maintained by the team which wraps the v2 MetaFunction API to support using object that can overwrote meta keys from parent routes.
